src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build.py

```
@@ -206,6 +206,16 @@ def _stream_logs(byte_size,
     if curr_bytes:
         print(curr_bytes.decode('utf-8', errors='ignore'))
 
+    build_status = _get_build_status(metadata).lower()
+    logger.debug("Build status was: '{}'".format(build_status))
+
+    if build_status == 'internalerror' or build_status == 'failed':
```

From the k-map is there any other condition - I might be missing something here. 
 build_status !== null && build_status.Status == "success" return 0 
Failure can return something like 
CLIError("Build status: '{}' with exit code ".format(build_status.status, build_status.exitCode)
I guess you might even be able to replace Complete with exitCode. 
Again I haven't through this through so better to have a way to remove complete with a single metadata field was the basic idea.

---

For CLI scenario, if users write a batch script and need to process build status (maybe retry for certain build failure), they will need to handle CLI exit code first and parse the last line of output anyway. In this case, exitCode won't provide more information than status.

A related question, for show-logs command, do you also want to raise the error?

---

@sajayantony statuses are the following. Queued is not possible here. Status codes are identical to this IMO. BTW, I'm not sure if we can even call `sys.exit(exit code #)` on the CLI though, so this exit code ends up being `1` for any failure. @djyou maybe can comment if we can get specific error codes for the CLI. I was told they don't like sys.exit before. :|

```
// Queued means the build has been queued but not picked up by a worker.
Queued BuildStatusEnum = "queued"

// InProgress means the build is being processed by a worker.
InProgress BuildStatusEnum = "inprogress"

// Successful means the build completed and was successful.
Successful BuildStatusEnum = "successful"

// InternalError means the build failed because of an internal error.
InternalError BuildStatusEnum = "internalerror"

// Failed means the build failed because of a general error.
Failed BuildStatusEnum = "failed"

// Canceled means the build was canceled.
Canceled BuildStatusEnum = "canceled"

// TimedOut means the build timed out and was unable to complete.
TimedOut BuildStatusEnum = "timedout"
```

---

@northtyphoon not sure if we want to error on show-logs, but that's the current behavior. To me it just adds more information; you can always discard the exit code if you don't plan to use it. Maybe we want a flag to disable errors though.

---

I doubt if CLI has a consistent way to output meaningful exit code and in most cases raising CLIError is the suggested approach. Even if we can exit with a code, I would question how easy it is for users to understand the meaning of it. Given that we end up exiting with the same code, I think this path can be simplified to something like 
`raise CLIError("Build exited with status: {}".format(build_status))` if not successful.